hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStage.java

```diff
@@ -750,6 +750,48 @@
             @Nonnull BiFunctionEx<T, T1, R> mapToOutputFn
     );
 
+    /**
+     * Attaches to both this and the supplied stage a inner-hash-joining stage and
+     * returns it. This stage plays the role of the <em>primary stage</em> in
+     * the hash-join. Please refer to the {@link com.hazelcast.jet.pipeline
+     * package javadoc} for a detailed description of the hash-join transform.
+     * <p>
+     * This sample joins a stream of users to a stream of countries and outputs
+     * a stream of users with the {@code country} field set:
+     * <pre>{@code
+     * // Types of the input stages:
+     * BatchStage<User> users;
+     * BatchStage<Map.Entry<Long, Country>> idAndCountry;
+     *
+     * users.innerHashJoin(
+     *     idAndCountry,
+     *     JoinClause.joinMapEntries(User::getCountryId),
+     *     (user, country) -> user.setCountry(country)
+     * )
+     * }</pre>
+     *
+     * <p>
```

Comment:
Please follow the formatting guidelines you can infer from the rest of the docs. We use text line length of 72 (excluding any indentation and `*`),  we don't use `</p>` and we don't indent the text.

Reply:
You should also update the Javadoc that this doc points to. The `com.hazelcast.jet.pipeline` Javadoc now explicitly states that "hash join is a left-outer join".